test/evp_extra_test.c

```diff
@@ -3076,6 +3076,71 @@ static int test_RSA_OAEP_set_null_label(void)
     return ret;
 }
 
+#ifndef OPENSSL_NO_DEPRECATED_3_0
+static int test_RSA_legacy(void)
+{
+    int ret = 0;
+    BIGNUM *p = NULL;
+    BIGNUM *q = NULL;
+    BIGNUM *n = NULL;
+    BIGNUM *e = NULL;
+    BIGNUM *d = NULL;
+    const EVP_MD *md = EVP_sha256();
+    EVP_MD_CTX *ctx = NULL;
+    EVP_PKEY *pkey = NULL;
+    RSA *rsa = NULL;
+
+    if (nullprov != NULL)
+        return TEST_skip("Test does not support a non-default library context");
+
+    if (!TEST_ptr(p = BN_dup(BN_value_one()))
+        || !TEST_ptr(q = BN_dup(BN_value_one()))
+        || !TEST_ptr(n = BN_dup(BN_value_one()))
+        || !TEST_ptr(e = BN_dup(BN_value_one()))
+        || !TEST_ptr(d = BN_dup(BN_value_one())))
+        goto err;
```

Comment:
p,q,n,e,d are all set to 1.

What are we actually testing here?

It does not seem to exercise anything related to the generation of rsa private-key data.

Are the legacy rsa APIs not tested elsewhere?

Reply:
The test is meant to exercise the legacy RSA key api (this was added on as this issue has evolved).  There is an issue referenced somewhere in this now massive thread indicating that RSA_set0_factos and RSA_set0_key werent working properly with the import and export api for EVP_KEY.  The intent of this test is to use those legacy apis to create an RSA key and then ensure that the key management api could export key data and re-import it.  EVP_DigestSignInit preforms both those operations, and since the initial issue was found using that api, its what I used in the test.

Reply:
Wouldnt a comment at the top of the function be nice to remember what/why the test was added. Might not be obvious in 6 months time,